myapp/app/models/user.rb

```diff
@@ -1,4 +1,38 @@
 class User < ApplicationRecord
+  before_save :downcase_email
+
   validates :name, presence: true, length: { minimum: 3, maximum: 20 }
+  validates :email, presence: true, format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i },
+                    uniqueness: { case_sensitive: true }
+  validates :password, presence: true, length: { minimum: 5, maximum: 20 }
+  has_secure_password
+
+  enum authoritys: { member: 0, adminer: 10, owner: 20 }
```

Comment:
```suggestion
  enum authorities: { member: 0, adminer: 10, owner: 20 }
```
タイポですかね

```suggestion
  enum roles: { member: 0, adminer: 10, owner: 20 }
```
でも良さそうですね

Reply:
ご指摘ありがとうございます！
命名ミスってますね。。
rolesに変更したいのですが、次のステップと大きくコンフリクトしそうなので、
次のステップで修正させていただければと思います:pensive: